lazy_theta_star_b/include/lazy_theta_star_b/lazy_theta_star_b.h

```diff
@@ -0,0 +1,176 @@
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Anshumaan Singh
+ *********************************************************************/
+#include<cmath>
+#include<iostream>
+#include<cstdlib>
+#include<cstring>
+#include <string>
+#include<queue>
+#include<algorithm>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "nav2_core/global_planner.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+
+namespace lazy_theta_star_b {
+    typedef int id;
+    using namespace std;
+
+#define SMALL 0.001
+#define INF_COST 10000000.0
+#define LETHAL_COST 100
+
+    template<typename ptsT>
+    struct pts {
+        ptsT x, y;
+    };
+
+    struct pos {
+        id pos_id;
+        double *g;
+        double *f;
+    };
+
+    struct tree_node {
+        int x, y;
+        double g = INF_COST;
+        double h = INF_COST;
+        id parentId;
+        int closed = 0;       //0 - unexplored, 1 - closed, -1 - open
+        double f = INF_COST;
+        int counter = 0;
+        int got_here = 0;
+    };
+
+
+    class LazyThetaStarB : public nav2_core::GlobalPlanner {
+
+    public:
+        
+        int lethal_cost = LETHAL_COST;
+
+        std::shared_ptr <tf2_ros::Buffer> tf_;
+        nav2_util::LifecycleNode::SharedPtr node_;
+        nav2_costmap_2d::Costmap2D *costmap_;
+        std::string global_frame_, name_;
+
+        double interpolation_dist;
+        int how_many_corners_;
+        pts<int> src_{}, dst_{};
+        
+        //stores the cell data
+        vector<tree_node> data;
+        
+        //stores the raw path given by the planner on the basis of their index in the vector data
+        vector<id> path;
+        nav_msgs::msg::Path global_path;
+        
+        // is the priority queue
+        vector<pos> pq;
+        int sizeX = 0;
+        int sizeY = 0;
+        int how_many_corners;
+        pts<int> src{}, dst{};
+
+
+        void configure(
+                rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+                std::string name, std::shared_ptr <tf2_ros::Buffer> tf,
+                std::shared_ptr <nav2_costmap_2d::Costmap2DROS> costmap_ros) override;
+
+        void cleanup() override;
+
+        void activate() override;
+
+        void deactivate() override;
+
+        nav_msgs::msg::Path createPlan(
+                const geometry_msgs::msg::PoseStamped &start,
+                const geometry_msgs::msg::PoseStamped &goal) override;
+
+
+        //The Line of Sight checking algorithm, takes in the points, and follows the line joining points
+        bool los_check(int x0, int y0, int x1, int y1);
```

Comment:
You should update any functions you can to have the inputs be `const references` and when const isn't possible, still a reference. Those copies will slow things down.

Reply:
Alright!